### ISSUE TITLE
max_open_trades Does not work when we use forceenter

### ISSUE THREAD

**Issue by ketank1000** [freqtrade/freqtrade#7489]
## Describe your environment

  * Operating system: MacOS 12.5.1
  * Python Version: 3.10.6
  * CCXT version: 1.92.52
  * Freqtrade Version: 2022.8
  * dry run : True
  
## Describe the problem:

I send my own signals to freqtrade to make trades, using forceexit and forceenter. If this calls are made simultaneously, freqtrade takes all the trades  even if the count is grater than max_open_trades.

### Steps to reproduce:

  1. set max_open_trades=3 and start the bot
  2. call forceenter simultaneously 10 times. 
  3. all trades will be taken
  
### Observed Results:

  * forceenter takes trades irrespective of max_open_trades
  * if new forceenter call comes, freqtrade should skip/through error if the count exceeds max_open_trades

  * When ran live it will through error if the there is no money in wallet after max_open_trade is completed. But I think this needs to be handled in the dry run as well (not tested this scenario).

**Comment by xmatthias** [comment]
`max_open_trade ` is respected by forceenter if it's used in a sane manner (Wait with the next request until the prior request returns).
The case you're making is an edge-case where you hammer the API with parallel requests (it's impossible to get to this state via telegram afaik)- so it becomes a race-condition - and the "> max open trade" check is made early on in the process - so no trade has been created yet - which could count against the "max_open_trades".


the only real way around this would be to prevent parallel calls  - so we'll reject all calls until the first call is finished and the trade is fully created.
I'm not entirely sure if that'd really be expected behavior though - as it would prevent cases where `max_open_trades` is high enough and you CAN do multiple parallel entries.

**Comment by ketank1000** [comment]
I would suggest to build a queueing mechanism, As we maintain our own DB, we can have worker to pick the job and execute it.

Right now I have implemented my own worker around flask API to handle parallel requests.

I liked the idea of rejecting the calls. It would be fast fix. Atleast it prevent the parallel requests.

**Comment by xmatthias** [comment]
the problem is - you kindof expect a response back - like "accepted" - or not - within reasonable time.

I currently don't really see this as a bug - as it's an edge usecase - which is actually caused by wrong usage of the API. 
when entering trades manually (not via some script) - you'll not encounter this.

**Comment by ketank1000** [comment]
Yes i accept that im not using the API right, but i dint find any other way to accept the calls, Even the webhook docs we have are just to send out messages and not accept the the message. If you have better solution please let us know.

Okay I will tell you my whole Architecture how I have made whole setup, I have my own strategy on tradingview where i have added multiple alerts (say around 100 cryptocurrencies). These messages are sent to my flask API. And flask api interprets the message and uses this scirpt [rest_obj](https://github.com/freqtrade/freqtrade/blob/develop/scripts/rest_client.py) to make the calls.

As you know behaviour of all cryptos are somewhat similer, the chances of sending request at same time are more (lets say when market starts falling). So at this point when market starts falling it takes 10-15 trades by the time `max_open_trades` reaches it limit.

Im not entering the trades manually, Entire setup is automated.

I will be happy to help out for this edge-case and raise PR.

**Comment by xmatthias** [comment]
to be perfectly honest - i think this should be handled in the client.

Every solution i can think of so far involves either not returning a "worked or didn't work" response directly (so you won't know if it worked) - or rejecting all calls until the first call went through.
Both solutions might work for your particular usecase - but are against the main usage of the API, which is freqUI (where users place trade manually, and would expect a response).

If you have any ideas how to approach this without introducing either of the above limitations let me know and we can discuss further.
Otherwise i'll leave this open for a bit if we have new ideas, but will then close it with an entry to the docs mentioning this as limitation.

**Comment by ketank1000** [comment]
Currently, I also do not see other option than implementing worker to executing task in queue, But that is too much amount of work for such edge case.

Another simple way to handle such scenario is, Can we add new feature to listen webhook calls and execute, by this way the API calls will be made in right way. Right now we just have webhook calls sending message, but we dont have any incoming calls.

This feature will be very useful for people.
* sending messages form tradingview like me
* sending messages from some AL/ML model
* Or if anyone want to scrap some data from internet and make calls

This will unleash the dependency on the strategy model we have.

**Comment by xmatthias** [comment]
I don't see how this would change behavior - even if the call is from a webhook directly - these calls could/would still be made in parallel (we work on the assumption that all pairs trigger at the same time) - so this won't do anything to help this cause.
It'd actually make this worse - as then you don't control the sender.

>  Right now we just have webhook calls sending message, but we dont have any incoming calls.

That's only partially correct - and i don't think you're aware how webhooks actually work / what they are.

We would have to decide on a format we expect the webhook message to be in - which will almost certainly be the same format we use for the API at this point (i don't see why this format should change). 
This may or may not correspond to the one from tradingview - but if we align to tradingview, it'll be incompatible with other providers, and is as such an endless endavor.

Technically - a rest API can also act as webhook receiver - assuming the sender can format the messages properly.
The API you build in the middle (the one which is causing this bug now) is also just an API that accepts messages, transforms them into the expected format, and then fires the call off again - so in reality, it's just transforming the message - nothing more.

Adding incoming webhooks will however also cause other problems - not least that we recommend to [never run the API directly on the internet](https://www.freqtrade.io/en/stable/rest-api/#configuration) - which is a recommendation we will not change - and would therefore render inbound webhook pointless.

**Comment by xmatthias** [comment]
Uppon closer inspection you're right, this setting is bypassed by forceentries.

as such, i suspect that a simple fix will suffice to fix this problem - at least in the broader sense (not the timing issue per se).

**Comment by ketank1000** [comment]
Thanks for fixing it !
